lib/sqlalchemy/ext/asyncio/engine.py

```diff
@@ -88,14 +91,12 @@ def begin(self) -> "AsyncTransaction":
         """Begin a transaction prior to autobegin occurring.
 
         """
-        self._sync_connection()
```

Comment:
this is here basically to do an error check, "if not self.sync_connection: raise.. " it doesn't need to be the same as _sync_connection() but it should be something

Reply:
Oh I thought this error check is done in `AsyncTransaction.start()` anyway, so I removed this early check. This also makes `await conn.begin()` consistent with `await engine.connect()` where the error is raised on `await`, not the awaitable-generating part, pls see also https://github.com/sqlalchemy/sqlalchemy/issues/5530#issuecomment-678897947.